tests/model_selection/splitters/maxmin_split.py

```diff
@@ -0,0 +1,182 @@
+
+    assert len(train_set) == 7
+    assert len(test_set) == 3
+
+    train_set, test_set = maxmin_train_test_split(
+        all_molecules, train_size=0.6, test_size=0.4, random_state=0
+    )
+
+    assert len(train_set) == 6
+    assert len(test_set) == 4
+
+
+def test_maxmin_valid_split_size(all_molecules):
+
+    train_set, valid_set, test_set = maxmin_train_valid_test_split(
+        all_molecules, train_size=0.7, test_size=0.2, valid_size=0.1, random_state=0
+    )
+
+    assert len(train_set) == 7
+    assert len(test_set) == 2
+    assert len(valid_set) == 1
+
+    train_set, valid_set, test_set = maxmin_train_valid_test_split(
+        all_molecules, train_size=0.5, test_size=0.2, valid_size=0.3, random_state=0
+    )
+
+    assert len(train_set) == 5
+    assert len(test_set) == 2
+    assert len(valid_set) == 3
+
+
+def test_seed_consistency_train_valid_test_split(all_molecules):
+
+    train_set_1, valid_set_1, test_set_1 = maxmin_train_valid_test_split(
+        all_molecules, train_size=0.7, valid_size=0.1, test_size=0.2, random_state=0
+    )
+
+    train_set_2, valid_set_2, test_set_2 = maxmin_train_valid_test_split(
+        all_molecules, train_size=0.7, valid_size=0.1, test_size=0.2, random_state=0
+    )
+
+    assert train_set_1 == train_set_2
+    assert valid_set_1 == valid_set_2
+    assert test_set_1 == test_set_2
+
+
+def test_seed_consistency_train_test_split(all_molecules):
+
+    train_set_1, test_set_1 = maxmin_train_test_split(
+        all_molecules, train_size=0.7, test_size=0.3, random_state=0
+    )
+
+    train_set_2, test_set_2 = maxmin_train_test_split(
+        all_molecules, train_size=0.7, test_size=0.3, random_state=0
+    )
+
+    assert train_set_1 == train_set_2
+    assert test_set_1 == test_set_2
+
+
+def test_seed_diversity_train_valid_test_split(all_molecules):
+
+    train_set_1, valid_set_1, test_set_1 = maxmin_train_valid_test_split(
+        all_molecules, train_size=0.7, valid_size=0.1, test_size=0.2, random_state=0
+    )
+
+    train_set_2, valid_set_2, test_set_2 = maxmin_train_valid_test_split(
+        all_molecules, train_size=0.7, valid_size=0.1, test_size=0.2, random_state=1
+    )
+
+    assert (
+        train_set_1 != train_set_2
+        or valid_set_1 != valid_set_2
+        or test_set_1 != test_set_2
+    )
+
+
+def test_seed_diversity_train_test_split(all_molecules):
+
+    train_set_1, test_set_1 = maxmin_train_test_split(
+        all_molecules, train_size=0.7, test_size=0.3, random_state=0
+    )
+
+    train_set_2, test_set_2 = maxmin_train_test_split(
+        all_molecules, train_size=0.7, test_size=0.3, random_state=1
+    )
+
+    assert train_set_1 != train_set_2 or test_set_1 != test_set_2
+
+
+def test_maxmin_train_test_split_return_molecules(all_molecules):
+    mols = [Chem.MolFromSmiles(smiles) for smiles in all_molecules]
+    train_set, test_set = maxmin_train_test_split(
+        mols, train_size=0.7, test_size=0.3, random_state=1, return_indices=False
```

Comment:
Why specify `return_indices=False` when it's false by default?

Reply:
Also don't use `random_state` in tests unless necessary. I would think that it's necessary to pass it here for determinism, when it's not

Reply:
done